core/src/main/scala/kafka/docker/KafkaDockerWrapper.scala

```diff
@@ -219,6 +231,68 @@ object KafkaDockerWrapper extends Logging {
     Files.write(filepath, properties.getBytes(StandardCharsets.UTF_8), mode)
   }
 
+  private def addToYaml(loggerFromEnv: Array[Logger], rootOption: Option[Root], filepath: Path): Unit = {
+    val path = filepath
+    if (!Files.exists(path)) {
+      Files.createFile(path)
+    }
+
+    val mapper = new ObjectMapper(new YAMLFactory().disable(Feature.WRITE_DOC_START_MARKER))
+      .configure(DeserializationFeature.ACCEPT_EMPTY_STRING_AS_NULL_OBJECT, true)
+      .setSerializationInclusion(JsonInclude.Include.NON_EMPTY)
+      .findAndRegisterModules();
+
+    val yaml = try {
+      mapper.readValue(filepath.toFile, classOf[Log4jConfiguration])
+    } catch {
+      case _: MismatchedInputException => new Log4jConfiguration
+      case e: RuntimeException => throw e
+    }
+    val config = yaml.getConfiguration
+
+    if (config == null && loggerFromEnv.isEmpty && rootOption.isEmpty) {
```

Comment:
we should remove this condition. otherwise, the script used by docker can't see the output log4j2 config ...

Reply:
it can be fixed by #18872